pkg/controller/shared_state.go

```diff
@@ -77,27 +77,32 @@ type SharedState struct {
 	crdClient                     crdClientset.Interface
 	conditionWatcher              *watcher.ConditionWatcher
 	azureDiskCSITranslator        csitranslator.InTreePlugin
+	driverUninstallState          *uint32
```

Comment:
Rather the store a pointer to the uninstall state, define an interface that defines a single method, `IsDriverUninstalling()`, to be implemented by the `CrdProvisioner`.